homeassistant/components/snapcast/media_player.py

```diff
@@ -22,72 +21,51 @@
     STATE_PLAYING,
     STATE_UNKNOWN,
 )
-import homeassistant.helpers.config_validation as cv
-from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers import config_validation as cv, entity_platform
 
-from . import (
+from .const import (
+    ATTR_LATENCY,
     ATTR_MASTER,
-    DOMAIN,
+    CLIENT_PREFIX,
+    CLIENT_SUFFIX,
+    DATA_KEY,
+    GROUP_PREFIX,
+    GROUP_SUFFIX,
     SERVICE_JOIN,
     SERVICE_RESTORE,
+    SERVICE_SET_LATENCY,
     SERVICE_SNAPSHOT,
     SERVICE_UNJOIN,
 )
 
 _LOGGER = logging.getLogger(__name__)
 
-DATA_KEY = "snapcast"
-
 SUPPORT_SNAPCAST_CLIENT = (
     SUPPORT_VOLUME_MUTE | SUPPORT_VOLUME_SET | SUPPORT_SELECT_SOURCE
 )
 SUPPORT_SNAPCAST_GROUP = (
     SUPPORT_VOLUME_MUTE | SUPPORT_VOLUME_SET | SUPPORT_SELECT_SOURCE
 )
 
-GROUP_PREFIX = "snapcast_group_"
-GROUP_SUFFIX = "Snapcast Group"
-CLIENT_PREFIX = "snapcast_client_"
-CLIENT_SUFFIX = "Snapcast Client"
-
-PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
```

Comment:
We still need the platform schema. That schema is validating the config yaml, not the service data.

Reply:
Oh! So when I was testing without the PLATFORM_SCHEMA, everything worked fine. I assume it worked because everything was already in the config; the problem is that it just wasn't being validated? Reading [this](https://developers.home-assistant.io/docs/configuration_yaml_index) but would like a bit more clarification.

Reply:
Yeah, platforms that are set up via config need a platform schema to a validate the config.